utils/prometheus/processor.py

```diff
@@ -0,0 +1,95 @@
+# (C) Datadog, Inc. 2017
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+import requests
+from utils.prometheus import parse_metric_family
+
+
+# message.type is the index in this array
+# see: https://github.com/prometheus/client_model/blob/master/ruby/lib/prometheus/client/model/metrics.pb.rb
+METRIC_TYPES = ['counter', 'gauge', 'summary', 'untyped', 'histogram']
+
+
+class PrometheusProcessor:
+    def __init__(self, conf):
+        # metrics_mapper is a dictionnary where the keys are the metrics to capture and the values are the corresponding metrics names to have in datadog.
+        self.metrics_mapper = {}
+        # send_histograms_buckets is used to specify if yes or no you want to send the buckets as tagged values when dealing with histograms.
+        self.send_histograms_buckets = True
+        # endpoint is the metrics endpoint to use to poll metrics from Prometheus
+        self.endpoint = 'http://localhost:10055/metrics'
+        # Conf from the agent
+        self.log = conf.log
+        self.gauge = conf.gauge
+
+    def process(self, **kwargs):
+        """
+        Polls the data from prometheus and pushes them as gauges
+        """
+        data = self.processor.poll()
+        for metric in parse_metric_family(data):
+            self.process_metric(metric, **kwargs)
+
+    def process_metric(self, message, **kwargs):
+        """
+        Handle a prometheus metric message according to the following flow:
+            - search self.metrics_mapper for a prometheus.metric <--> datadog.metric mapping
+            - call check method with the same name as the metric
+            - log some info if none of the above worked
+        """
+        try:
+            if message.name in self.metrics_mapper:
+                self._submit_metric(self.metrics_mapper[message.name], message)
+            else:
+                getattr(self, message.name)(message, **kwargs)
+        except AttributeError as err:
+            self.log.debug("Unable to handle metric: {} - error: {}".format(message.name, err))
+
+    def poll(self):
+        """
+        Polls the metrics from the prometheus endpoint
+        """
+        headers = {
+            'accept': 'application/vnd.google.protobuf; proto=io.prometheus.client.MetricFamily; encoding=delimited',
+            'accept-encoding': 'gzip',
+        }
+        req = requests.get(self.endpoint, headers=headers)
+        req.raise_for_status()
+        return req.content
+
+    def _submit_gauge(self, metric_name, val, metric, custom_tags=[]):
```

Comment:
Oh wait, ok, the comment explanation just arrived to my brain, took a long time, sorry!
Will do.